Connect.vb

```diff
@@ -144,5 +149,81 @@ Public Class Connect
+    ''' </summary>
+    ''' <param name="toolWindowCaption">
+    ''' String you need to put as a header on the window.
+    ''' </param>
+    ''' <param name="toolWindowGuid">
+    ''' This uniquely identified a particular window can be used to store 
+    ''' information of it like size, position, etc.
+    ''' </param>
+    ''' <param name="toolWindowUserControl">
+    ''' Windows to host.
+    ''' </param>
+    Private Function CreateToolWindow(ByVal toolWindowCaption As String, ByVal toolWindowGuid As String,
+      ByVal toolWindowUserControl As UserControl) As Window
+        Dim userControlObject As Object = Nothing
+        Dim userControlHost As UserControlHost
+        Dim _SummaryWindow As Window
+        Dim progId As String
+        ' IMPORTANT: ensure that you use the same ProgId value used in the ProgId attribute of the UserControlHost 
+        progId = "MyVBAAddin.UserControlHost"
+        _SummaryWindow = _VBE.Windows.CreateToolWindow(_AddIn, progId, toolWindowCaption, toolWindowGuid, userControlObject)
+        userControlHost = DirectCast(userControlObject, UserControlHost)
+        _SummaryWindow.Visible = True
+        userControlHost.AddUserControl(toolWindowUserControl)
+        Return _SummaryWindow
+    End Function
+
+    ''' <summary>
+    ''' Implementation of the click event of the button.
+    ''' </summary>
+    ''' <param name="Ctrl"></param>
+    ''' <param name="CancelDefault"></param>
+    Private Sub _Button_Click(Ctrl As Microsoft.Office.Core.CommandBarButton,
+      ByRef CancelDefault As Boolean) Handles _Button.Click
+        'If the button is clicked first time.
+        If count = 1 Then
+            'Make a copy of the file active in excel.
+            My.Computer.FileSystem.CopyFile(_VBE.ActiveVBProject.FileName, "D:\GoogleDrive.xlsm", True)
+            Dim letsTry As uploadFileToDrive = New uploadFileToDrive()
+            'Upload the file on drive.
+            fileId = letsTry.UploadFile("D:\GoogleDrive.xlsm")
+            'Delete that file.
+            My.Computer.FileSystem.DeleteFile("D:\GoogleDrive.xlsm")
+            'If no error ocurred then fileId is not empty.
+            If fileId <> "" Then
+                lines = hittingEndPoint.callSheetsAPI(fileId)
+            Else
+                Exit Sub
+            End If
+        End If
+        'if the data list count item is not zero then all Api in the file is not supported.
+        If lines.Count <> 0 Then
+            Dim userControlObject As Object = Nothing
+            Dim userControlToolWindow1 As UserControlToolWindow1
+            Try
+                    'If button is clicked the first time then window needs to initialize.
+                If _ApiSummaryWindow Is Nothing Then
+                    userControlToolWindow1 = New UserControlToolWindow1()
+                    _ApiSummaryWindow = CreateToolWindow("Report of API used in this project", "0EB93108-D229-4F6F-82C5-0B96AFFBB9C5", userControlToolWindow1)

```

Comment:
What is this 0EB93108-D229-4F6F-82C5-0B96AFFBB9C5?

Reply:
Defined in CreateToolWindow() function summary, What does this argument mean.